NBitcoin/Protocol/NodesCollection.cs

```diff
@@ -79,7 +98,8 @@ public MessageProducer<IncomingMessage> MessageProducer
 			}
 		}
 
-		ConcurrentDictionary<Node, Node> _Nodes = new ConcurrentDictionary<Node, Node>();
+		ConcurrentDictionary<Node, Node> _Nodes = new ConcurrentDictionary<Node, Node>(new NodeComparer());
```

Comment:
Nice. Is there a place where the nodes are actually compared before they're added to the dictionary?

Reply:
The changed take effect in the "TryAdd" call of the Add method.

Reply:
public bool Add(Node node)
		{
			if(node == null)
				throw new ArgumentNullException("node");
			if(_Nodes.TryAdd(node, node))
			{
				node.MessageProducer.AddMessageListener(bridge);
				OnNodeAdded(node);
				return true;
			}
			return false;
		}

Reply:
Without a comparer the TryAdd has no way to know how to compare for equality. It would likely default to comparing object pointers.

Reply:
As seen in this case not always, unless it has a comparer...